core/src/main/scala/akka/persistence/jdbc/state/scaladsl/JdbcDurableStateStore.scala

```diff
@@ -86,56 +87,90 @@ class JdbcDurableStateStore[A](
   def deleteObject(persistenceId: String): Future[Done] =
     db.run(queries._delete(persistenceId).map(_ => Done))
 
-  def currentChanges(tag: String, offset: Offset): Source[DurableStateChange[A], NotUsed] = offset match {
-    case NoOffset      => makeSourceFromStateQuery(tag, None)
-    case APSequence(l) => makeSourceFromStateQuery(tag, Some(l))
-    case _             => ??? // should not reach here
+  def currentChanges(tag: String, offset: Offset): Source[DurableStateChange[A], NotUsed] =
+    currentChangesByTag(tag, offset.value)
+
+  def changes(tag: String, offset: Offset): Source[DurableStateChange[A], NotUsed] =
+    _changesByTag(tag, offset.value, terminateAfterOffset = None)
+
+  private def currentChangesByTag(tag: String, offset: Long): Source[DurableStateChange[A], NotUsed] = {
+    Source
+      .futureSource(maxJournalSequence().map { maxOrderingInDb =>
+        _changesByTag(tag, offset, terminateAfterOffset = Some(maxOrderingInDb))
+      })
+      .mapMaterializedValue(_ => NotUsed)
+  }
+
+  private def _currentChangesByTag(
+      tag: String,
+      from: Long,
+      batchSize: Long,
```

Comment:
`batchSize` here but in `changesByTag` it's rather the max number of results to collect?

Reply:
`max` changed to `batchSize`

Reply:
But then it is used as max in the actual query (limiting the results rather than deciding how large each batch is)?

Reply:
@johanandren The `max` is used to indicate the maximum number of rows that the query will fetch. That's equal to the `batchSize` in our case. Hence keeping the name uniform as `batchSize` looks ok I thought.

Reply:
Ah, ok, thanks for clarifying!